openshift/openshift-ansible-contrib
# Upscaling OpenShift application nodes

#### What does this PR do?
It scales up number of application nodes by a value predefined by the user.
When value is not defined, the current number of app nodes is increased by 1.
When there is no deployment to scale up, the value becomes the target number of app nodes.

#### How should this be manually tested?
Note: Always remove `hosts` file from your inventory when deleting a deployment.

**Test it on an existing deployment**:
1) Run `oc get nodes -l autoscaling=app` on your master node.
- If the result is empty, it means that you haven't completed a deployment where this label gets set. Therefore, you have to delete your current deployment and run a new end to end deployment that sets the autoscaling label first (which is done automatically, so no extra steps during the development are required).
- If the result displays current application nodes, you're all set.
2) Now that you have met the prerequisites, run the `openshift-ansible-contrib/playbooks/provisioning/openstack/scale-up.yaml` playbook.
- In order to scale up your deployment by a custom amount, add `-e increment_by=<number greater or equal to 1>` when executing `ansible-playbook`.
- You can customise the path to your `openshift-ansible` directory by adding `-e openshift_ansible_dir=<path to openshift-ansible>`. The value must be an absolute path and cannot contain '/' at the end.
3) Check that the deployment and playbook pre/post-verifications were successful. Try assigning a workload to the newly created nodes.

**Test it without an existing deployment**:
1) If you have a deployment, delete it.
2) Run the `openshift-ansible-contrib/playbooks/provisioning/openstack/scale-up.yaml` playbook.
- In order to create a custom amount of nodes (1 by default), add `-e increment_by=<number greater or equal to 1>` when executing `ansible-playbook`.
- You can customise the path to your `openshift-ansible` directory by adding `-e openshift_ansible_dir=<path to openshift-ansible>`. The value must be an absolute path and cannot contain '/' at the end.
3) Check that the deployment and playbook pre/post-verifications were successful. Try assigning a workload to the newly created nodes.

#### Is there a relevant Issue open for this?
--

#### Who would you like to review this?
cc: @tomassedovic @bogdando @tzumainn  PTAL
